sdk/export/trace/tracetest/test_test.go

```diff
@@ -0,0 +1,57 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracetest
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"go.opentelemetry.io/otel/sdk/export/trace"
+)
+
+var _ trace.SpanBatcher = (*InMemorySpanBatcher)(nil)
```

Comment:
Ah, can we move this next to the struct itself?

Reply:
to the struct? you probably mean in the same file.

I read once that these kind of checks are better in test files (and I liked their reasons, and would recommend the same):
* Avoid unnecessary dependency if the file does not depend on the package that the interface implements.

Reply:
> Avoid unnecessary dependency if the file does not depend on the package that the interface implements.

I'm confused, it does already depend on the package that the interface implements.

Reply:
It just happen to be in this case because SpanData is there, but I follow that rule all the time as a best practice.

Reply:
Will move to follow your conventions, but trying to tell you about this idea.

Reply:
> ...but trying to tell you about this idea.

Ah, I see. Thanks for the info. :+1: 

I'm a fan of keeping relevant piece of the code close to one another so a developer doesn't have to go looking. It also seems from my experience to group these compile time checks near the struct (though it is also usually the case to not leak interfaces). Good to keep in mind as this project is refined.

Reply:
Done.